cmd/cluster/resize/controlplane_node.go

```diff
@@ -94,6 +95,11 @@ func (o *controlPlane) New() error {
 		return err
 	}
 
+	supportedInstanceTypes := []string{"m5.4xlarge", "m5.8xlarge", "m5.12xlarge", "m5.16xlarge", "m5.24xlarge"}
```

Comment:
What do you think about defining this as a `var` in `cmd/cluster/resize/cmd.go` with a map for `infra` and `control-plane`, then this validation at the top of `New()` as the first thing we check to fail as early as possible if the values are bad?

Sample map
```go
var supportedInstanceTypes = map[string][]string{
	"controlplane": {"m5.4xlarge", "m5.8xlarge", "m5.12xlarge", "m5.16xlarge", "m5.24xlarge"},
	"infra":        {"r5.4xlarge", "r5.8xlarge", "r5.12xlarge", "r5.16xlarge", "r5.24xlarge"},
}
```

Reply:
And then add/extend the tests to show it failing validation when the input values are not valid :)